apps/ejabberd/src/mod_mam_mnesia_prefs.erl

```diff
@@ -117,172 +120,145 @@ stop_muc(Host) ->
 %% ----------------------------------------------------------------------
 %% Internal functions and callbacks
 
--spec get_behaviour(DefaultBvr :: mod_mam:archive_behaviour(),
-        _Host :: ejabberd:server(), ArcId :: mod_mam:archive_id(),
-        LocJID :: ejabberd:jid(), RemJID :: ejabberd:jid())
-            -> mod_mam:archive_behaviour().
-get_behaviour(DefaultBehaviour, _Host, _ArcID,
+-spec get_behaviour(Default :: behaviour(), Host :: ejabberd:server(),
```

Comment:
Not a big deal, but regarding style, wouldn't it be better to format it like this, if we have a spec so long that it spans multiple lines?

```erlang
-spec get_behaviour(DefaultBh, Host, ArcId, LocJID, RemJID) -> Result when
      DefaultBh :: mod_mam:archive_behaviour(),
      Host :: ejabberd:server(),
      ArcId :: mod_mam:archive_id(),
      LocJID :: ejabberd:jid(),
      RemJID :: ejabberd:jid(),
      Result :: mod_mam:archive_behaviour().
```

On one hand _de gustibus non disputandum est_, but on the other it seems much more readable to me that way. For specs that make it into a single line _inline_ types are OK from my point of view.